pkg/apiserver/certificate/cacert_controller_test.go

```diff
@@ -0,0 +1,566 @@
+				clientset.PrependReactor("update", "configmaps", func(action cgtesting.Action) (handled bool, ret runtime.Object, err error) {
+					return true, &corev1.ConfigMap{}, errors.New("error updating configmap")
+				})
+			},
+			expectedErrMsg: "error updating ConfigMap antrea-ca: error updating configmap",
+		},
+		{
+			name: "fail to get antrea-ca ConfigMap",
+			prepareReactor: func(clientset *fakeclientset.Clientset) {
+				clientset.PrependReactor("get", "configmaps", func(action cgtesting.Action) (handled bool, ret runtime.Object, err error) {
+					return true, &corev1.ConfigMap{}, errors.New("error getting configmap")
+				})
+			},
+			expectedErrMsg: "error getting ConfigMap antrea-ca: error getting configmap",
+		},
+	}
+	caConfig := &CAConfig{
+		ServiceName:     AntreaServiceName,
+		PairName:        "antrea-controller",
+		CAConfigMapName: "antrea-ca",
+	}
+	secureServing := genericoptions.NewSecureServingOptions().WithLoopback()
+	var err error
+	caConfig.SelfSignedCertDir, err = os.MkdirTemp("", "antrea-self-signed")
+	if err != nil {
+		t.Fatalf("Unable to create temporary directory: %v", err)
+	}
+	defer os.RemoveAll(caConfig.SelfSignedCertDir)
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			clientset := fakeclientset.NewSimpleClientset()
+			if tt.existingCM != nil {
+				clientset = fakeclientset.NewSimpleClientset(tt.existingCM)
+			}
+			aggregatorClientset := fakeaggregatorclientset.NewSimpleClientset()
+			apiExtensionClient := fakeapiextensionclientset.NewSimpleClientset()
+			caContentProvider, _ := generateSelfSignedCertificate(secureServing, caConfig)
+			tt.prepareReactor(clientset)
+
+			controller := newCACertController(caContentProvider, clientset, aggregatorClientset, apiExtensionClient, caConfig)
+			err := controller.syncConfigMap([]byte("abc"))
+			if tt.expectedErrMsg != "" {
+				require.ErrorContains(t, err, tt.expectedErrMsg)
+				return
+			}
+			require.NoError(t, err)
+			cm, err := clientset.CoreV1().ConfigMaps("kube-system").Get(context.Background(), "antrea-ca", metav1.GetOptions{})
+			require.NoError(t, err)
+			require.Equal(t, "abc", cm.Data["ca.crt"])
+		})
+	}
+}
+
+func TestSyncAPIServices(t *testing.T) {
+	tests := []struct {
+		name               string
+		existingAPIService *apiregistrationv1.APIService
+		prepareReactor     func(clientset *fakeaggregatorclientset.Clientset)
+		expectedErrMsg     string
+	}{
+		{
+			name: "sync API Service successfully",
+			existingAPIService: &apiregistrationv1.APIService{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "v1alpha1.stats.antrea.io",
+				},
+				Spec: apiregistrationv1.APIServiceSpec{
+					Service: &apiregistrationv1.ServiceReference{
+						Namespace: "kube-system",
+						Name:      "antrea",
+					},
+					CABundle: []byte("def"),
+				},
+			},
+			prepareReactor: func(clientset *fakeaggregatorclientset.Clientset) {},
+		},
+		{
+			name:           "fail to sync API Service",
```

Comment:
```suggestion
			name:           "fail to get APIService",
```
sync covers get, create, update, not same level as the other test name

Reply:
Done